batched_blas/batched_blas.md

```diff
@@ -325,6 +325,110 @@ have a natural representation without them.
 
 1. Should broadcast input parameters be represented by `mdspan` with one less rank (than the batched case would normally require), or by some other method?
 
+# Wording Sketch
+
+In P1673 the wording for individual functions is very simple:
+
+```c++
+template<class ExecutionPolicy,
+         in-matrix InMat,
+         in-vector InVec,
+         out-vector OutVec>
+void matrix_vector_product(ExecutionPolicy&& exec,
+                           InMat A,
+                           InVec x,
+                           OutVec y);
+```
+
+*Effects:* Computes $y = Ax$
+
+This wording relies on exposition only concepts _`in-matrix`_, _`in-vector`_ and _`out-vector`_ given by:
+
+```c++
+template<class T>
+concept @_in-vector_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  T::rank() == 1;
+
+template<class T>
+concept @_out-vector_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  T::rank() == 1 &&
+  is_assignable_v<typename T::reference, typename T::element_type> &&
+  T::is_always_unique();
+```
+
+We propose adding new overloads of the function with new exposition only concepts, accounting for broadcasting of input arguments:
+
+```c++
+template<class T>
+concept @_batched-in-vector_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  (T::rank() == 1 || T::rank() == 2);
+
+template<class T>
+concept @_batched-out-vector_@ = // exposition only
+  @_is-mdspan_@<T>::value &&
+  T::rank() == 2 &&
+  is_assignable_v<typename T::reference, typename T::element_type> &&
+  T::is_always_unique();
+```
+
+The batched _`batched-out-***`_ and _`batched-inout-***`_ concepts strictly increase the rank requirement by one, _`in-***`_ concepts allow the original rank and one higher.
+
+In addition to these new exposition only concepts we need some helper functions:
+
+```c++
+template<@_batched-in-vector_@ InVec>
+requires(InVec::rank()==1)
+auto get_batch(InVec v, int /*batch*/) { return v; }
+
+template<@_batched-in-vector_@ InVec>
+requires(InVec::rank()==2)
+auto get_batch(InVec v, int batch) { return submdspan(v, batch, full_extent); }
+
+template<@_batched-out-vector_@ OutVec>
+auto get_batch(OutVec v, int batch) { return submdspan(v, batch, full_extent); }
```

Comment:
Assuming that you define `@_batched-in-matrix_@` somewhere, I don't see how you can stir A passed as rank-2 and x passed at rank-2 towards different overloads of `get_batch`.

Reply:
yeah you are right, needs a name `get_batch_vector`, `get_batch_matrix` ...